kernel/src/log_segment.rs

```diff
@@ -112,13 +126,245 @@ impl LogSegment {
     }
 }
 
+/// Builder for [`LogSegment`] from from `start_version` to `end_version` inclusive
+pub(crate) struct LogSegmentBuilder<'a> {
+    fs_client: &'a dyn FileSystemClient,
+    table_root: &'a Url,
+    start_checkpoint: Option<CheckpointMetadata>,
+    start_version: Option<Version>,
+    end_version: Option<Version>,
+    /// When `commit_files_sorted_ascending` is set to `true`, the commit files are sorted in
+    /// ascending order. Otherwise if it is set to `false`, the commit files are sorted in
+    /// descending order. This is set to `false` by default.
+    commit_files_sorted_ascending: bool,
```

Comment:
this is for the builder right, so it should be an instruction, suggest:
```suggestion
    sort_commit_files_ascending: bool,
```